Fix broken test descriptions

smowton opened the pull request:
These two tests accidentally passed one of their required outputs as a command-line argument.

tautschnig reviewed (changes requested):
Acting against all rules on how to be a good reviewer: more work, please.

It's great that you have noticed those problems (I'm actually wondering how), but the reason those tests still passed is that the test descriptions do not include
```
^EXIT=0$
^SIGNAL=0$
```
Being the terrible person that I am: can you please, as part of this PR, add this to all *.desc files that are missing it, with those saying "VERIFICATION FAILED" using `EXIT=10`instead. Thank you!

smowton committed:
Fix tests with missing EXIT or SIGNAL tests

Some tests had previously been passing despite actually causing a crash, due
to the required output being too loosely specified. This ensures the bare minimum:
that every test has an expected EXIT code and SIGNAL result.

The codes suggested were taken from the test's current output, and only applied for
CORE tests, but hand inspection suggests these choices are reasonable.

smowton commented:
@tautschnig done

tautschnig commented on regression/ansi-c/Forward_Declaration2/test.desc:
```diff
@@ -1,6 +1,7 @@
 CORE
 main.c
 
+^EXIT=1$
```
I believe this needs to be `^EXIT=(64|1)$` to cope with Windows.

tautschnig commented on regression/ansi-c/Incomplete_Type1/test.desc:
```diff
@@ -1,6 +1,7 @@
 CORE
 main.c
 
+^EXIT=1$
```
I believe this needs to be `^EXIT=(64|1)$` to cope with Windows.

tautschnig commented on regression/ansi-c/static2/test.desc:
```diff
@@ -2,5 +2,7 @@ CORE
 main.c
 main2.c --function foo
 ^main symbol `foo' is ambiguous$
+^EXIT=1$
```
I believe this needs to be `^EXIT=(64|1)$` to cope with Windows.

tautschnig commented on regression/ansi-c/static3/test.desc:
```diff
@@ -2,5 +2,7 @@ CORE
 main.c
 main2.c --function foo
 ^main symbol `foo' is ambiguous$
+^EXIT=1$
```
I believe this needs to be `^EXIT=(64|1)$` to cope with Windows.

smowton replied:
Do you know why?

smowton replied:
Ah never mind I got it, goto-cc vs. goto-cl

tautschnig reviewed (commented):
Thank you very much for all the extra work! Just some requests in ansi-c to cope with a portability problem.

smowton commented:
@tautschnig changes applied, CI is passing except for CodeBuild which is opaque to me and AFAIK not yet intended to work

tautschnig commented:
This now needs a rebase. Maybe @kroening can in the meantime explain what's going wrong with CodeBuild?

smowton commented:
@tautschnig done, let's see if CodeBuild complains again

tautschnig commented:
regression/cbmc-java/lambda1 seems to conflict with symex-lazy-loading, if my diagnosis is correct.

smowton commented:
Yup, and the new test revealed it. Adding an entry point so the test can work under either mode now.

smowton committed:
Make test lambda1 compatible with symex-driven loading

Previously it didn't have an entry-point, and so relied on the normal load process
loading all functions. The added entry-point mimics that behaviour when loading is
on-demand.

smowton commented:
Fixed.
